back/engines/commercial/user_custom_fields/spec/models/user_custom_fields/representativeness/ref_distribution_spec.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe UserCustomFields::Representativeness::RefDistribution do
+  subject(:ref_distribution) { build(:ref_distribution) }
+
+  describe 'factory' do
+    it { is_expected.to be_valid }
+  end
+
+  it { is_expected.to belong_to(:custom_field) }
+  it { is_expected.to have_many(:values).through(:custom_field).source(:custom_field_options) }
+  it { is_expected.to validate_uniqueness_of(:custom_field_id).case_insensitive }
+  it { is_expected.to validate_presence_of(:distribution) }
+
+  it 'validates that the distribution has at least 2 options', :aggregate_failures do
```

Comment:
This test does not verify "at least 2 options". It verifies "exactly 2 options". So I think an extra expectation should be added for a distribution with 3 options.

Reply:
This spec is meant to be an (simplified) equivalent of something like:
```ruby
it { is_expected.to validate_length_of(:distribution).is_at_least(2).with_short_message('must have at least 2 options.') }
```
which doesn't work in this case because it tries with multiple test values that depends on the column type. In the case of a `jsonb`, those test values also include json arrays which are not valid/realistic distributions and it is freaking out other validation logic.

How would you go about that?

Reply:
Oops, sorry missed the part about:
> So I think an extra expectation should be added for a distribution with 3 options.

Actually, I chose to test the boundary here (2 -> OK, 1 -> KO). Why do you think adding an expectation with 3 options would help? 3 seems pretty arbitrary to me.

Reply:
Without the case for 3, I can change the validation to verify for exactly 2 options, like `validates :distribution, presence: true, length: { is: 2, message: 'must have exactly 2 options.' }`,  and the test would still pass.

Reply:
Devil's advocate: if I follow the same kind of reasoning, I can change the validation to verify for either 2 or 3 options: `validates :distribution, presence: true, length: {minimum: 2, maximum: 3}`. So, we would have to add an expectation to check that 4 options is allowed. Repeat the induction step, and now we need an expectation for all natural numbers >= 2.